src/main/scala/org/scalacheck/ScalaCheckFramework.scala

```diff
@@ -34,13 +34,19 @@ private abstract class ScalaCheckRunner extends Runner {
       case _ => false
     }
     if (countTestSelectors == 0) rootTask(taskDef)
-    else checkPropTask(taskDef)
+    else checkPropTask(taskDef, single = true)
   }
 
   def serializeTask(task: Task, serializer: TaskDef => String) =
     serializer(task.taskDef)
 
-  def tasks(taskDefs: Array[TaskDef]): Array[Task] = taskDefs.map(rootTask)
+  def tasks(taskDefs: Array[TaskDef]): Array[Task] = {
+    val isForked = taskDefs.exists(_.fingerprint().getClass.getName.contains("ForkMain"))
```

Comment:
This is a bit icky, but I'm not aware of an official way to detect if we're forked or not. /cc @dwijnand @eed3si9n